task8/src/main/java/PostCollection.java

```diff
@@ -0,0 +1,157 @@
+                }
+            } else if (pair.getKey().equals(KEY_VALIDATE_UNTIL)) {
+                if (pair.getValue() instanceof Date) {
+                    posts.stream()
+                            .filter(post -> post.getValidateUntil().equals(pair.getValue()))
+                            .forEach(filteredPosts::add);
+                }
+            } else if (pair.getKey().equals(KEY_HASHTAG)) {
+                if (pair.getValue() instanceof String) {
+                    posts.stream()
+                            .filter(post -> Arrays.asList(post.getHashTags()).contains(pair.getValue()))
+                            .forEach(filteredPosts::add);
+                }
+            } else if (pair.getKey().equals(KEY_RATING)) {
+                if (pair.getValue() instanceof Integer) {
+                    posts.stream()
+                            .filter(post -> post.getRating() == (int) pair.getValue())
+                            .forEach(filteredPosts::add);
+                }
+            } else if (pair.getKey().equals(KEY_DISCOUNT)) {
+                if (pair.getValue() instanceof Integer) {
+                    posts.stream()
+                            .filter(post -> post.getDiscount() == (int) pair.getValue())
+                            .forEach(filteredPosts::add);
+                }
+            }
+        }
+        if (filterConfig.size() == 0) {
+            filteredPosts = new ArrayList(posts);
+        }
+        if (top > filteredPosts.size()) {
+            top = filteredPosts.size();
+        }
+        if (skip >= filteredPosts.size()) {
+            skip = 0;
+        }
+        if (skip + top > filteredPosts.size()) {
+            return filteredPosts.subList(skip, filteredPosts.size());
+        } else {
+            return filteredPosts.subList(skip, skip + top);
+        }
+    }
+
+
+    public Post getPost(int id) {
+        for (Post post : posts) {
+            if (post.getId() == id) {
+                return post;
+            }
+        }
+        return null;
+    }
+
+    public boolean validatePost(Post post) {
+        for (Post currentPost : posts) {
+            if (currentPost.getId() == post.getId()) {
+                return false;
+            }
+        }
+        if (post.getDescription() == null || post.getDescription().length() > 200) {
+            return false;
+        }
+        if (post.getAuthor() == null) {
+            return false;
+        }
+        if (!post.getCreationDate().before(post.getValidateUntil())) {
+            return false;
+        }
+        if (post.getDiscount() < 0 || post.getDiscount() > 100) {
+            return false;
+        }
+        return true;
+    }
+
+    public boolean addPost(Post post) {
+        if (validatePost(post)) {
+            posts.add(post);
+            return true;
+        } else {
+            return false;
+        }
+    }
+
+    public boolean editPost(int id, Post filterConfig) {
+        Post post = getPost(id);
```

Comment:
For the current and the above methods, nullity checks are missing. Methods are called on by-reference objects, without checking them are not null. This is generally not good for the methods that are `public`. With public methods, we can never guarantee what comes from outside